packages/hop-node/src/watchers/xDomainMessageRelayWatcher.ts

```diff
@@ -115,15 +149,74 @@ class xDomainMessageRelayWatcher extends BaseWatcher {
       return
     }
 
+    const chainSlug = this.chainIdToSlug(await this.bridge.getChainId())
     const watcher = this.watchers[chainSlug]
     if (!watcher) {
       logger.warn(`exit watcher for ${chainSlug} is not implemented yet`)
       return
     }
 
-    logger.debug(`handling commit tx hash ${commitTxHash} from ${destinationChainId}`)
+    logger.debug(`handling commit tx hash ${commitTxHash} to ${destinationChainId}`)
     await watcher.handleCommitTxHash(commitTxHash, transferRootId, logger)
   }
+
+  async checkConfirmableTransferRoots (transferRootId: string) {
+    const dbTransferRoot = await this.db.transferRoots.getByTransferRootId(transferRootId) as ExitableTransferRoot
+    if (!dbTransferRoot) {
+      throw new Error(`transfer root db item not found, root id "${transferRootId}"`)
+    }
+
+    const { transferRootHash, destinationChainId, totalAmount, committedAt } = dbTransferRoot
+
+    const logger = this.logger.create({ root: transferRootId })
+    const isTransferRootIdConfirmed = await this.l1Bridge.isTransferRootIdConfirmed(
+      destinationChainId,
+      transferRootId
+    )
+    if (isTransferRootIdConfirmed) {
+      logger.warn('Transfer root already confirmed')
+      await this.db.transferRoots.update(transferRootId, {
+        confirmed: true
+      })
+      return
+    }
+
+    if (this.dryMode) {
+      this.logger.warn(`dry: ${this.dryMode}, skipping confirmRootsViaWrapper`)
+      return
+    }
+
+    await this.db.transferRoots.update(transferRootId, {
+      sentConfirmTxAt: Date.now()
+    })
+
+    logger.debug(`handling confirmable transfer root ${transferRootHash}, destination ${destinationChainId}, amount ${totalAmount.toString()}, committedAt ${committedAt}`)
+    await this.confirmRootsViaWrapper([{
+      rootHash: transferRootHash,
+      destinationChainId,
+      totalAmount,
+      rootCommittedAt: committedAt
+    }])
+  }
+
+  async confirmRootsViaWrapper (rootData: ConfirmRootData[]): Promise<void> {
+    const rootHashes: string[] = []
+    const destinationChainIds: number[] = []
+    const totalAmounts: BigNumber[] = []
+    const rootCommittedAt: number[] = []
+    for (const data of rootData) {
+      rootHashes.push(data.rootHash),
+      destinationChainIds.push(data.destinationChainId),
+      totalAmounts.push(BigNumber.from(data.totalAmount)),
```

Comment:
I think `totalAmount` should be a big number already

```suggestion
      totalAmounts.push(data.totalAmount),
```